File: samples/Ruby/partial-function-application.rb
fs = proc { |f, s| s.map &f }
f1 = proc { |n| n * 2 }
f2 = proc { |n| n ** 2 }
fsf1 = fs.curry[f1]
fsf2 = fs.curry[f2]

[0..3, (2..8).step(2)].each do |e|
  p fsf1[e]
  p fsf2[e]
end

File: samples/Haskell/partial-function-application.hs
fs = map
f1 = (* 2)
f2 = (^ 2)

fsf1 = fs f1
fsf2 = fs f2

main :: IO ()
main = do
  print $ fsf1 [0, 1, 2, 3] -- prints [0, 2, 4, 6]
  print $ fsf2 [0, 1, 2, 3] -- prints [0, 1, 4, 9]
  print $ fsf1 [2, 4, 6, 8] -- prints [4, 8, 12, 16]
  print $ fsf2 [2, 4, 6, 8] -- prints [4, 16, 36, 64]
